xee/helpers.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Helper functions for grid parameters."""
+import math
+
+import ee
+from pyproj import Transformer
+from rasterio.transform import Affine
+import shapely
+from shapely.ops import transform
+from shapely.geometry import box
+from typing import TypedDict, Tuple, Union
+
+
+TransformType = Tuple[float, float, float, float, float, float]
+ShapeType = Tuple[int, int]
+ScalingType = Union[float, Tuple[float, float]]
+
+
+class PixelGridParams(TypedDict):
+    crs: str
+    crs_transform: TransformType
+    shape_d2: ShapeType
+
+
+def set_scale(
+    crs_transform: TransformType,
+    scaling: ScalingType,
+  ) -> list:
+  """Update the CRS transform's scale parameters."""
+  match scaling:
+    case int(xy_scale) | float(xy_scale):
+      crs_transform[0] = xy_scale
+      crs_transform[4] = xy_scale
+    case (int(x_scale) | float(x_scale), int(y_scale) | float(y_scale)):
+      crs_transform[0] = x_scale
+      crs_transform[4] = y_scale
+    case _:
+      raise TypeError
+  affine_transform = Affine(*crs_transform)
```

Comment:
Is it necessary to cast as Affine object? I ask because it gets converted to list before being returned - we're not e.g., using any affine methods on the object.

Reply:
It's probably good for validating, but we should try to use `affine` library directly, instead of from `rasterio.transform` to avoid the sometimes troublesome dependency.